File: docs/snippets/go/cache/disk_cache.go
```go title="disk_cache.go"
package main

import (
	"fmt"
	"log"

	"github.com/xberg-io/xberg/packages/go"
)

func main() {
	useCache := true
	namespace := "documents"
	ttl := uint64(7 * 86400)

	config := xberg.ExtractionConfig{
		UseCache:       &useCache,
		CacheNamespace: &namespace,
		CacheTTLSecs:   &ttl,
	}

	fmt.Println("First extraction (will be cached)...")
	input1 := xberg.ExtractInputFromURI("document.pdf")
	result1, err := xberg.Extract(*input1, config)
	if err != nil {
		log.Fatalf("extract failed: %v", err)
	}
	if len(result1.Results) > 0 {
		fmt.Printf("  - Content length: %d\n", len(result1.Results[0].Content))
	}

	fmt.Println("\nSecond extraction (from cache)...")
	input2 := xberg.ExtractInputFromURI("document.pdf")
	result2, err := xberg.Extract(*input2, config)
	if err != nil {
		log.Fatalf("extract failed: %v", err)
	}
	if len(result2.Results) > 0 {
		fmt.Printf("  - Content length: %d\n", len(result2.Results[0].Content))
	}

	if len(result1.Results) > 0 && len(result2.Results) > 0 {
		fmt.Printf("\nResults are identical: %v\n", result1.Results[0].Content == result2.Results[0].Content)
	}
}
```

File: docs/snippets/rust/cache/disk_cache.rs
```rust title="disk_cache.rs"
use xberg::{extract, ExtractionConfig, ExtractInput};

#[tokio::main]
async fn main() -> xberg::Result<()> {
    let path = std::env::args()
        .skip(1)
        .find(|a| !a.is_empty() && a != "--")
        .unwrap_or_else(|| "document.pdf".to_string());

    // Enable caching (default: true). The Rust crate uses an internal disk cache.
    let config = ExtractionConfig {
        use_cache: true,
        ..Default::default()
    };

    println!("First extraction (will be cached)...");
    let output1 = extract(ExtractInput::from_uri(&path), &config).await?;
    println!("  - Content length: {}", output1.results[0].content.len());

    println!("\nSecond extraction (from cache when available)...");
    let output2 = extract(ExtractInput::from_uri(&path), &config).await?;
    println!("  - Content length: {}", output2.results[0].content.len());

    println!("\nResults are identical: {}", output1.results[0].content == output2.results[0].content);

    Ok(())
}
```
